Automattic/wp-calypso
# Copy Site: Generate destination site slug based on source slug

#### Proposed Changes

When I copied `test.wpcomstaging.com`, the new site was `<my username><random number>.wpcomstaging.com`. I would've expected it to be `test12.wpcomstaging.com` or similar (original slug with a random hash appended).

#### Testing Instructions



* Select an atomic site under `/sites`
* Click on the ellipsis menu and select `Copy Site`
* Continue with the steps till you navigate to the `/checkout` screen.
* Navigate back to `/sites`
* Ensure that there is a new site created named similar to your previous atomic site. (Eg: If you had `test.wpcomstaging.com`, look for something like `test12.wordpress.com`)

![copy_site_fix_slug_1](https://user-images.githubusercontent.com/497103/212702186-551d72ef-4138-4f75-9d4d-65a844ee7a88.png)

![copy_site_fix_slug_2](https://user-images.githubusercontent.com/497103/212702214-819d5272-c472-454d-a073-331a9e9f41e4.png)

![copy_site_fix_slug_3](https://user-images.githubusercontent.com/497103/212702241-804ac437-7b01-47d7-bf93-b7f5f663a1c2.png)


#### Pre-merge Checklist





Related to https://github.com/Automattic/dotcom-forge/issues/1496
